src/tests/ftest/deployment/disk_failure.py

```diff
@@ -143,8 +143,22 @@ def test_disk_fault_to_normal(self):
                 self.log.info("  %s: %s", key, device[key])
             try:
                 self.dmg_command.hostlist = NodeSet(host)
+                # Set the device as faulty
                 get_dmg_response(
-                    self, self.dmg_command.storage_replace_nvme,
-                    old_uuid=device["uuid"], new_uuid=device["uuid"])
+                    self, self.dmg_command.storage_set_faulty, uuid=device["uuid"])
+                # Replace the device with same uuid.
+                error_count = 0
+                for _ in range(10):
+                    data = self.dmg_command.storage_replace_nvme(old_uuid=device["uuid"],
+                                                                 new_uuid=device["uuid"])
+                    time.sleep(5)
+                    if data['error'] or len(data['response']['host_errors']) > 0:
+                        error_count = error_count + 1
+                    else:
+                        error_count = 0
+                        break
+                if error_count > 0:
+                    self.fail("replace_nvme command failed: {}".
+                              format(data['response']['host_errors']))
```

Comment:
Why sleep between calling `storage_replace_nvme` and checking for errors? Couldn't we sleep after checking the error?

Reply:
Agree... Sleep could be done later. I don't think it is critical. Presently, we have to wait atleast above 10 seconds for things go work. I can fix this issue later (if needed). Being a manual test, I think it should  be ok for now.